examples/dynamo/scripts/redundant_setval_c.py

```diff
@@ -0,0 +1,78 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Laboratory
+
+'''File containing a PSyclone transformation script for the Dynamo0.3
+API to apply redundant computation to halo depth 1 for all instances
+of loops that iterate over dofs and contain the setval_c builtin.
+
+'''
+from psyclone.transformations import Dynamo0p3RedundantComputationTrans
+
+ITERATION_SPACES = ["dofs"]
+KERNEL_NAMES = ["setval_c"]
+DEPTH = 1
+
+
+def trans(psy):
+    '''PSyclone transformation script for the dynamo0.3 API to apply
+    redundant computation into the level 1 halo generically to all
+    loops that iterate over dofs and exclusively contain the setval_c
+    builtin. The reason for choosing this particular builtin is that
+    this builtin only writes to data so will not cause any additional
+    halo exchanges, or increases in halo exchange depth, through
+    redundant computation.
+
+    '''
+    rc_trans = Dynamo0p3RedundantComputationTrans()
+
+    transformed = 0
+
+    for invoke in psy.invokes.invoke_list:
+        schedule = invoke.schedule
+        for loop in schedule.loops():
+            if loop.iteration_space in ITERATION_SPACES:
+                # we may have more than one kernel in the loop so
+                # check that all of them are in the list of accepted
+                # kernel names
+                setcalls = True
+                for call in loop.calls():
+                    if not call.name in KERNEL_NAMES:
+                        setcalls = False
```

Comment:
Again, you could use `continue` here and do away with `setcalls` entirely.

Reply:
Again, I'm afraid I don't see how to do this. I've added a break to avoid unnecessary iterations but if there is a better solution please let me know.

Reply:
Again, I think I failed to see the outer loops. Ignore me :-)